pennylane/plugins/tests/__init__.py

```diff
@@ -47,4 +32,163 @@
+operating system. To find the location, you can use the :func:`~.get_device_tests` function:
+
+>>> from pennylane.plugins.tests import get_device_tests
+>>> get_device_tests()
+
+The pl-device-test CLI
+----------------------
+
+Alternatively, PennyLane provides a command line interface for invoking the device tests.
+
+.. code-block:: console
+
+    pl-device-test --device default.qubit --shots 1234 --analytic False
+
+Within Python
+-------------
+
+Finally, the tests can be invoked within a Python session via the :func:`~.test_device`
+function:
+
+>>> from pennylane.plugins.tests import test_device
+>>> test_device("default.qubit")
+
+For more details on the available arguments, see the :func:`~.test_device` documentation.
+
+Functions
+---------
 """
+# pylint: disable=import-outside-toplevel
+import argparse
+import pathlib
+import sys
+
+
+def get_device_tests():
+    """Returns the location of the device integration tests."""
+    return str(pathlib.Path(__file__).parent.absolute())
+
+
+def test_device(device, analytic=None, shots=None, skip_ops=True, pytest_args=None):
+    """Run the device integration tests using an installed PennyLane device.
+
+    Args:
+        device (str): the name of the device to test
+        analytic (bool): Whether to run the device in analytic mode (where
+            expectation values and probabilities are computed exactly from the quantum state)
+            or non-analytic/"stochastic" mode (where probabilities and expectation
+            values are *estimated* using a finite number of shots.)
+            If not provided, the device default is used.
+        shots (int): The number of shots/samples used to estimate expectation
+            values and probability. Only takes affect if ``analytic=False``. If not
+            provided, the device default is used.
+        skip_ops (bool): whether to skip tests that use operations not supported
+            by the device
+        pytest_args (list[str]): additional PyTest arguments and flags
+
+    **Example**
+
+    >>> from pennylane.plugins.tests import test_device
+    >>> test_device("default.qubit")
+    ================================ test session starts =======================================
+    platform linux -- Python 3.7.7, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
+    rootdir: /home/josh/xanadu/pennylane/pennylane/plugins/tests, inifile: pytest.ini
+    plugins: flaky-3.6.1, cov-2.8.1, mock-3.1.0
+    collected 86 items
+    xanadu/pennylane/pennylane/plugins/tests/test_gates.py ..............................
+    ...............................                                                       [ 70%]
+    xanadu/pennylane/pennylane/plugins/tests/test_measurements.py .......sss...sss..sss   [ 95%]
+    xanadu/pennylane/pennylane/plugins/tests/test_properties.py ....                      [100%]
+    ================================= 77 passed, 9 skipped in 0.78s ============================
```

Comment:
Just a heads up, I get these two warnings:

```
.../lib/python3.6/site-packages/_pytest/config/__init__.py:935: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: flaky
    self._mark_plugins_for_rewrite(hook)

.../lib/python3.6/site-packages/_pytest/config/__init__.py:935
PytestAssertRewriteWarning: Module already imported so cannot be rewritten: pytest_mock
    self._mark_plugins_for_rewrite(hook)
```

Reply:
It turns out this was my fault, you are not supposed to use `pytest.main()` to invoke tests 😆 I've switched this to a simple `subprocess.call()`, and these warnings disappear